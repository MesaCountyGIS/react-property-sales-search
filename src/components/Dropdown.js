import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../styles/Dropdown.css';

  class Dropdown extends Component {
      constructor(props){
          super(props);
          this.state = {
              display: false,
              listItems: props.option,
              baseClass: props.baseclass
          };
      }

componentDidMount = (x) =>{
    document.addEventListener('click', this.showHide, false);
}

componentWillUnmount = () =>{
   document.removeEventListener('click', this.showHide, false)
 }

selectNode = (node) =>{
    if (node.target.className !== 'heading') {
        this.props.handleChange[0](node.target.innerHTML); //updateRecordCountButton()
        this.props.handleChange[1](); //getCount() from MoreFilters.js
        // this.props.handleChange[0](node.target.innerHTML);
    }
}

showHide = (event) =>{
    if (event.target.className.indexOf(this.state.baseClass) > -1) {
        this.setState({display: !this.state.display});
    } else if (event.target.className.indexOf(this.state.baseClass) <= -1 &&
        event.target.className !== "heading" &&
        this.state.display) {
        this.setState({display: false});
    }
}


    render() {
        //ulDisplay is outside of the ul style tag so it can be spread outside
        //along with the customStyles.dropNav object
        const ulDisplay = {
            "display":this.state.display? "block":"none"
        }
      return (
          <div style={{...this.props.customStyles.ulContainer}}
              className={`dropdown ${this.state.baseClass}`}>{this.props.value}
              <span className={`dropdown ${this.state.baseClass}`}>&#9660;</span>
              <ul style={{...ulDisplay, ...this.props.customStyles.dropNav}}
                  className='drop-nav display' onClick={this.selectNode}>
                  {this.state.listItems.map(item =>{
                      return <li key={item.value} className={item.type} >{item.value}</li>
                  })}
              </ul>
          </div>
      );
    }
  }

  Dropdown.propTypes = {
      value: PropTypes.string,
      option: PropTypes.array.isRequired,
      baseclass: PropTypes.string,
      customStyles: PropTypes.object.isRequired
  }

  Dropdown.defaultProps = {
      value: 'Select...',
      baseclass: 'base-class',
      customStyles: {dropdown:{},dropNav:{},ulContainer:{}}
  }


export default Dropdown;
