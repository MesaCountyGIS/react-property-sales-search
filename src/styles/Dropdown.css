.dropdown {
  display: inline-block;
    position: relative;
    cursor: pointer;
    text-align:left;
    /*padding-left: 1em !important;*/
    top: -0.4em;
    font-weight:bold;
    font-size:0.8em;
    height:2.6em;
    width:17em;
}

span.dropdown {
  margin-right:1em;
  float:right;
  width:auto;
  top:0.1em;
}

.drop-nav {
  list-style:none;
  text-align:left;
  line-height:1.9em;
    cursor: default;
    position: absolute;
    top: 1.5em;
    background: white;
    border-radius: 0 0 7px 7px;
    box-shadow: 5px 5px 20px #888888;
    width: 16em;
    padding-left:0.5em;
    z-index: 100;
}

.drop-nav li {
    cursor:pointer;
    margin-left:1em;
    float: left;
    width: 93.5%;
}

.drop-nav li:hover {
    background: rgb(229, 229, 229);
}

.drop-nav li:last-child {
    border-bottom-right-radius: 7px;
}

.drop-nav li.heading {
  margin-left:0;
  font:bold;
  font-size:120%;
  color: #000000;
}

.drop-nav li.heading:hover {
  cursor:default;
  background:none;
}
