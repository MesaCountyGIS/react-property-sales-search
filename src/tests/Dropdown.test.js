import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import {stubConsoleError} from './utils/test-utils';
import Dropdown from '../components/Dropdown';

describe("<Dropdown />", ()=>{
    stubConsoleError();

    const testContent = [
         {type: 'heading', value: 'Improved Sales', label: 'Improved Sales'},
         {type: 'item', value: 'Agricultural', label: 'Agricultural' },
         {type: 'item', value: 'Commercial/Industrial', label: 'Commercial/Industrial' },
         {type: 'item', value: 'Agricultural Vacant', label: 'Agricultural Vacant' },
    ]

    test('renders without crashing', () => {
        const tree = renderer.create(
            <Dropdown
                baseclass='test-dropdown'
                option={testContent}
                placeholder='Select Test Data'
                value={'Select Test Data'}/>
      ).toJSON();

      sinon.assert.notCalled(console.error)
      expect(tree).toMatchSnapshot();
    });

    test('UL shows when dropdown clicked', () => {
        const wrapper = mount(
            <Dropdown
                baseclass='test-dropdown'
                option={testContent}
                placeholder='Select Test Data'
                value={'Select Test Data'}
            />
      )

      wrapper.instance().showHide({target:{classname: 'header'}});
    //   wrapper.instance().changeMax({target:{value: 5000}});
    //   expect(wrapper.nodes[0].state.minValue).toEqual(500);
      expect(wrapper.nodes[0].state.isShown).toEqual(5000);
    });

});
