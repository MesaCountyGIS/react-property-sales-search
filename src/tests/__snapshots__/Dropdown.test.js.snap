// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> renders without crashing 1`] = `
<div
  className="dropdown test-dropdown"
  style={Object {}}
>
  Select Test Data
  <span
    className="dropdown test-dropdown"
  >
    â–¼
  </span>
  <ul
    className="drop-nav display"
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <li
      className="heading"
    >
      Improved Sales
    </li>
    <li
      className="item"
    >
      Agricultural
    </li>
    <li
      className="item"
    >
      Commercial/Industrial
    </li>
    <li
      className="item"
    >
      Agricultural Vacant
    </li>
  </ul>
</div>
`;
